### ISSUE TITLE
Doesn't work (for me) in ie11

### ISSUE THREAD

**Issue by Garionion** [jerrybendy/url-search-params-polyfill#36]
When I am trying this in ie11 I got the following:
`[Vue warn]: Error in created hook: "ReferenceError: "URLSearchParams" is undefined"`
`[object Error]{description: ""URLSearchP...", message: ""URLSearchP...", name: "ReferenceEr...", number: -2146823279, stack: "ReferenceEr..."}` (expandable to the Object)

I really don't know what i am doing wrong.

I am using Vue, Typescript, Vue Bootstrap, Vuex, Vue Router.
I am using URLSearchParams like this in my main.ts

```javascript
import Vue from 'vue';
import App from './App.vue';
import router from './router';
import store from './store';
import BootstrapVue from 'bootstrap-vue';
import VueMoment from 'vue-moment';
import Datepicker from 'vuejs-datepicker';
import GridLoader from 'vue-spinner/src/GridLoader.vue';
import 'url-search-params-polyfill';

import 'bootstrap/dist/css/bootstrap.css';
import 'bootstrap-vue/dist/bootstrap-vue.css';
[…]

const app = new Vue({
	data: { loading: false },
	router,
	store,
	created() {
		setParams();
	},
	render: (h) => h(App),
}).$mount('#app');

[…]

function setParams() {
	new URLSearchParams(window.location.search);
	const urlParams = new URLSearchParams(window.location.search);
// already failed
	[…]

}

```

_Originally posted by @Garionion in https://github.com/jerrybendy/url-search-params-polyfill/issues/30#issuecomment-444449982_

**Comment by jerrybendy** [comment]
Hi,
I don't know what happened in you code.
You can try to let the `import 'url-search-params-polyfill';` as the first sentence, before `import Vue from 'vue';`. I can't make sure if it works or not, just try it.

**Comment by jerrybendy** [comment]
I just tried this polyfill in IE11 again, and it works well. I think maybe your other codes or dependencies have used `URLSearchParams` class before you `import` it.

**Comment by Garionion** [comment]
I don't think so, because i got the error the first time after i add this piece of code. However, i am very thankful that you tried to help. I replaced this little piece with some regex - for now. If I ever get this polyfill to work at my use case, I will switching back.

**Comment by jerrybendy** [comment]
OK, regex is better if you only want to get a part of query string.

https://gist.github.com/jerrybendy/d9f76553ca42907cc95bb342e87beae2

7 lines of codes only  😬

**Comment by QuantumInformation** [comment]
This also not working for us in ie11 with a new create react app

**Comment by FadiAboMsalam** [comment]
+1 it also doesn't work in IE11

**Comment by jerrybendy** [comment]
I can do nothing for it, I even can not reproduce this issue. It will be better if you could make a demo to reproduce it

**Comment by rahul-rautwar** [comment]
I am using this polyfill, its working fine for Edge v15 but not on IE 11,
I am getting this error msg in console: "**Error: unsupported BodyInit type**"
Can someone suggest me how can I made it to work on IE11

**Comment by jerrybendy** [comment]
@rahul-rautwar  This is a [known issue](https://github.com/jerrybendy/url-search-params-polyfill#known-issues) when using this polyfill with `fetch`. You can fellow README to add a `Content-Type` header when you `fetch`ing.

**Comment by rahul-rautwar** [comment]
yes I am passing content-type header in my fetch options parameter,
headers: {
        'Content-Type': 'application/x-www-form-urlencoded;charset=UTF-8'
 }

I am getting this error msg in console of IE11: "Error: unsupported BodyInit type"

**Comment by jerrybendy** [comment]
@rahul-rautwar Could you make a demo to repeat this issue?

**Comment by chris-fran** [comment]
We just went through the same. It doesn't seem to work unless you have the Symbol polyfill installed on your own.
